ResearchSuiteUI/ResearchSuiteUI/iOS/Step View Controllers/RSDActiveStepViewController.swift

```diff
@@ -0,0 +1,137 @@
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import UIKit
+
+open class RSDActiveStepViewController: RSDStepViewController {
+
+    @IBOutlet open var countdownLabel: UILabel?
+    @IBOutlet open var instructionLabel: UILabel?
+    @IBOutlet open var countdownDial: RSDCountdownDial?
+    @IBOutlet open var progressLabel: UILabel?
+    @IBOutlet open var unitLabel: UILabel?
+
+    /**
+     Formatter for the countdown label.
+     */
+    lazy open var countdownFormatter : DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+        formatter.allowedUnits = [.minute, .second]
+        formatter.unitsStyle = .positional
+        formatter.zeroFormattingBehavior = [ .pad ]
+        return formatter
+    }()
+    
+    open override var countdown: Int {
+        didSet {
+            countdownLabel?.text = countdownFormatter.string(from: TimeInterval(countdown))
+        }
+    }
+
+    open override func speak(instruction: String, timeInterval: TimeInterval) {
+        super.speak(instruction: instruction, timeInterval: timeInterval)
+        instructionLabel?.text = instruction
+    }
+
+    open override func start() {
+        super.start()
+        startProgressAnimation()
+    }
+    
+    private func startProgressAnimation() {
+        guard self.isVisible, let stepDuration = self.activeStep?.duration, let uptime = self.startUptime
+            else {
+                return
+        }
+        
+        // Animate in 60 second intervals. This will result in a smoother animation
+        // while for a long-running step it will result in callbacks at intervals that
+        // allow for any drift.
+        let maxInterval = stepDuration > 60.0 ? 10.0 : stepDuration
```

Comment:
The comment says 60 second intervals, but this looks like if the step lasts over 60 seconds then it animates in 10 second intervals...?